### DISCUSSION TITLE
Looking for advice on a strange issue

### DISCUSSION THREAD

**Discussion by eyalabraham** [erichelgeson/BlueSCSI-v1#72] [Q&A]
Hi everyone!

I recently built a BlueSCSI kit which worked ok until some strange issues started and not it does not seem to work any more.
Hope I am not too verbose but I tried to do a much troubleshooting as a could before posting here.

I am using a Mac Plus, so needed an external USB power supply. Everything work ok, except that only one image would be recognize even if two valid ones were on the card.

This got me to a format reformat cycle, and at some point, not sure when, no images were recognized by the Mac. The floppy/question-mark blinks on the screen, and the BlueSCSI blinks green LED. Here is what I managed to test so far:
- BlueSCSI recognized the images on the card and assigned them correctly to SCSI IDs (LOG.txt file is there and correct)
- Measured voltages, 5.0v from the external supply
- All images work on emulators Basilisk 2 as well as mini vMAC
- Mac boots from a floppy if I insert one
- I reprogrammed the Blue Pill _outside of the board_ and wrote some code to test all IO pins for in/out; everything verifies ok. The issue is that the test was done with pins in Push Pull mode and not Open Drain, so this test passed.
- I also enabled internal Mac TERMPWR with a diode; things work buy the voltage dropped by internal and external board diode is to large so I'm back to external power supply.
- Rechecked all soldering points under magnifying glass and tested for shorts and breaks.

I've exhausted my options at this point and looking for any advice you might have. If all fails I'll buy a new Blue Pill?

Thanks in advance for any advice,
EA

**Comment by erichelgeson** [comment]
Hi EA - we should be able to help you out. Few comments/ideas:

> Everything work ok, except that only one image would be recognize even if two valid ones were on the card.

This commonly happens when mixing images with different SCSI drivers. Ensure each image you use is formatted with the same driver. Also the Mac Plus can not use the LIDO drivers if you had any image with them.

> All images work on emulators Basilisk 2 as well as mini vMAC

This is odd - mini vmac does not work with BlueSCSI images - vmac only works with HFS partitions. This has me thinking it's an issue with the image. BlueSCSI needs to have everything a real disk has - HFS Partition, Partition map, and SCSI Driver partition. Basilisk II can manage with either.

>If all fails I'll buy a new Blue Pill?

Nothing you've said so far has indicated that there's an issue with the BluePIll or BlueSCSI. Have you gone through the Troubleshooting wiki page? Specifically doing a full overwrite format with the "SD Card Formatter" Utility? I would also grab the Mac Plus specific images from the Usage page, as well as a Blank formatted with Apples drivers.

Since you have the ability to flash the bluepill you can also try flashing the latest main branch - lots of fixes in there that will be in a release soon (such as the ability to format without a patched apple formatting utility)

**Reply by eyalabraham** [reply]
@erichelgeson thank you for the suggestions above. I had never thought about the disk image format as I figured that if it was created by one of the emulators then it was good to go.

This is what I did: I created a blank image file with 'dd' on my Linux machine (after formatting the SD card with the appropriate SW on a Win VM I have), then proceeded to format the disk image with mini vMac, and then installed System 6 on it (also through mini vMAC)

I already have the latest code from you git hub repo flashed so I will try your suggestion to grab the Mac Plus specific images from the Usage page, as well as a Blank formatted with Apples drivers, and report back.

**Reply by erichelgeson** [reply]
Just for some more info on mini vmac and scsi disks https://www.gryphel.com/c/mail/v13.html#m309

**Reply by eyalabraham** [reply]
I wanted to get back with some results, and I am happy to report that I was successful after following the advice about the image types! I think I get the nuances now. thanks you @erichelgeson 

The RaSCSI-Boot-6.0.8 referenced from the BlueSCSI wiki works perfectly (as ID 6), but my Mac goes into a reboot loop if I add another drive on another ID (less than 6). Maybe I need to press shift while booting because the drivers(?) don't agree with the system I have but that's for another day (maybe tomorrow ;-)

I ended up putting two blank drives and loading OS from some floppies I have. Happy Mac.

**Reply by erichelgeson** [reply]
The rascsi boot images use lido for the scsi driver so that is expected with the plus. The mega link in the wiki has a mac plus image. Glad you got it working!
